src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -309,6 +313,18 @@ public void pulse() {
             damage(1, DamageCause.SUFFOCATION);
         }
 
+        // fire damage
+        if (getLocation().getBlock().getType() == Material.FIRE) {
+            damage(1, DamageCause.FIRE);
```

Comment:
The `pulse` method is executed **once per tick**, which is every 50ms. This check would damage the entity 20 times per second if they step in fire. Also, the `setFireTicks` call below would increment the fire duration by 180 ticks every tick it isn't on fire.

I think the overall logic needs some reviewing

Reply:
The fireTicks is only incremented when the entity has just recently exited the fire block, so by assigning False to stoodInFire we won't have a fireTicks counter incrementing by 180 with every tick.  Sorry about the confusion, I tested this on a live server where the damage in the fire and out of the fire proved successful.  Please let me know if I should make any changes.

Reply:
Hey, it's my bad for the confusion. After putting second thought to this, the logic does make sense 😊

Although you can simplify the `} else { if (stoodInFire) {` branches to an else-if statment.

Reply:
Thank you for the feedback though!  It seems I will have to do some more work in this implementation.